src/sentry/api/endpoints/rule_snooze.py

```diff
@@ -126,3 +124,53 @@ def post(self, request: Request, project, rule_id) -> Response:
             serialize(rule_snooze, request.user, RuleSnoozeSerializer()),
             status=status.HTTP_201_CREATED,
         )
+
+    def delete(self, request: Request, project, rule_id) -> Response:
+        if not features.has("organizations:mute-alerts", project.organization, actor=None):
+            raise AuthenticationFailed(
+                detail="This feature is not available for this organization.",
+                code=status.HTTP_401_UNAUTHORIZED,
+            )
+
+        serializer = RuleSnoozeValidator(data=request.data)
+        if not serializer.is_valid():
+            return Response(serializer.errors, status=status.HTTP_400_BAD_REQUEST)
+
+        data = serializer.validated_data
+        rule = self.get_rule(rule_id)
+        user_id = request.user.id if data.get("target") == "me" else None
+
+        if not can_edit_alert_rule(rule, project.organization, user_id, request.user):
+            raise AuthenticationFailed(
+                detail="Requesting user cannot mute this rule.", code=status.HTTP_401_UNAUTHORIZED
+            )
+
+        issue_alert_rule = rule if self.rule_type == RuleType.ISSUE_ALERT.value else None
+        metric_alert_rule = rule if self.rule_type == RuleType.METRIC_ALERT.value else None
+
+        try:
+            rulesnooze = RuleSnooze.objects.get(
+                user_id=user_id,
+                rule=issue_alert_rule,
+                alert_rule=metric_alert_rule,
+                owner_id=request.user.id,
+                until=data.get("until"),
+            )
+        except RuleSnooze.DoesNotExist:
+            return Response(
+                {"detail": "This rulesnooze object doesn't exist."},
+                status=status.HTTP_404_NOT_FOUND,
+            )
+
+        rulesnooze.delete()
+        return Response(status=status.HTTP_204_NO_CONTENT)
+
+
+@region_silo_endpoint
+class RuleSnoozeEndpoint(BaseRuleSnoozeEndpoint):
+    rule_type = RuleType.ISSUE_ALERT.value
+
+
+@region_silo_endpoint
+class MetricRuleSnoozeEndpoint(BaseRuleSnoozeEndpoint):
+    rule_type = RuleType.METRIC_ALERT.value
```

Comment:
I just realised we could take this a step further and have this be `rule_model` and `rule_field`. Then you could fetch the model like `rule_model.objects.get(id=x)` and create like 
```
kwargs = {rule_field: rule}
RuleSnooze.objects.get_or_create(
            user_id=user_id,
            defaults={
                "owner_id": request.user.id,
                "until": data.get("until"),
                "date_added": datetime.datetime.now(),
            },
            **kwargs
        )
```

Up to you if you think that's worthwhile